src/ol/source/stamensource.js

```diff
@@ -89,36 +89,71 @@ ol.source.StamenProviderConfig = {
  */
 ol.source.Stamen = function(options) {
 
-  var i = options.layer.indexOf('-');
-  var provider = i == -1 ? options.layer : options.layer.slice(0, i);
-  goog.asserts.assert(provider in ol.source.StamenProviderConfig);
-  var providerConfig = ol.source.StamenProviderConfig[provider];
-
-  goog.asserts.assert(options.layer in ol.source.StamenLayerConfig);
-  var layerConfig = ol.source.StamenLayerConfig[options.layer];
-
+  /**
+   * @type {string} url URL
+   * @private
+   */
   var root = ol.IS_HTTPS ? 'https://stamen-tiles-{a-d}.a.ssl.fastly.net/' :
       'http://{a-d}.tile.stamen.com/';
-  var url = goog.isDef(options.url) ? options.url :
-      root + options.layer + '/{z}/{x}/{y}.' +
-      layerConfig.extension;
+  this.rootUrl_ = goog.isDef(options.url) ? options.url : root;
+
+  /**
+   * Layer
+   * @type {string}
+   * @private
+   */
+  this.layer_ = options.layer;
 
   goog.base(this, {
     attributions: ol.source.Stamen.ATTRIBUTIONS,
     crossOrigin: 'anonymous',
-    maxZoom: providerConfig.maxZoom,
-    // FIXME uncomment the following when tilegrid supports minZoom
-    //minZoom: providerConfig.minZoom,
-    opaque: layerConfig.opaque,
-    tileLoadFunction: options.tileLoadFunction,
-    url: url
+    tileLoadFunction: options.tileLoadFunction
   });
+  this.setLayer(this.layer_);
 
 };
 goog.inherits(ol.source.Stamen, ol.source.XYZ);
 
 
 /**
+ * @param {string} layer
+ * @api stable
```

Comment:
Please use the `@api` annotation without the stability level.  We can mark this as stable after the initial release of the feature.  In addition, please add documentation for this method.  It should have a one sentence plain text summary in the description (before the first `@param`).  And then you can add additional detail (like links to valid Stamen layer names if appropriate) using markdown in additional sentences.